openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -2280,137 +2281,197 @@ def _detect_undefined_stereo(cls, rdmol, err_msg_prefix="", raise_warning=False)
                 raise UndefinedStereochemistryError(msg)
 
     @staticmethod
-    def _flip_rdbond_direction(rdbond, paired_rdbonds):
-        """Flip the rdbond and all those paired to it.
-
-        Parameters
-        ----------
-        rdbond : rdkit.Chem.Bond
-            The Bond whose direction needs to be flipped.
-        paired_rdbonds : Dict[Tuple[int], List[rdkit.Chem.Bond]]
-            Maps bond atom indices that are assigned a bond direction to
-            the bonds on the other side of the double bond.
+    def _constrain_end_directions(
+        *values: int, bond_indices: List[int], flip_direction: Dict[int, bool]
+    ) -> bool:
+        """A constraint applied when mapping global E/Z stereochemistry into local RDKit
+        bond directions that ensures that the 'left' bonds point in opposite directions
+        (i.e. one has to be up and one has to be down) and likewise for the 'right' bonds
         """
-        from rdkit import Chem
 
-        # The function assumes that all bonds are either up or down.
-        supported_directions = {Chem.BondDir.ENDUPRIGHT, Chem.BondDir.ENDDOWNRIGHT}
+        unique_values = {
+            (value if not flip_direction[i] else 1 - value)
+            for i, value in zip(bond_indices, values)
+        }
```

Comment:
This is a bit too compact to understand/debug easily - It would be good to either unpack this into a few lines or have a comment reminding the reader of what the variables contain.